kyo-pure/shared/src/main/scala/kyo/defers.scala

```diff
@@ -0,0 +1,25 @@
+package kyo
+
+import kyo.core.*
+import kyo.core.internal.*
+
+abstract private[kyo] class Defer[T, S] extends Suspend[Const[Unit], Unit, T, S & Defers]:
```

Comment:
This new effect provides suspension similarly to `IOs` but it's designed to handle only pure computations without side effects.